src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -203,10 +203,24 @@ trait SyntheticMethods extends ast.TreeDSL {
       if (tests.isEmpty) {
         thatTest(eqmeth)
       } else {
-        thatTest(eqmeth) AND Block(
-          ValDef(otherSym, thatCast(eqmeth)),
-          AND(tests: _*)
-        )
+        def andy = {
+          val otherDef = ValDef(otherSym, thatCast(eqmeth))
+          if (tests.size < 23) Block(otherDef, AND(tests: _*))
+          else {
+            val ifs = ListBuffer[Tree]()
+            ifs += otherDef
+            val groupit = tests.grouped(22)
+            while (groupit.hasNext) {
+              val g = groupit.next()
+              if (groupit.hasNext)
+                ifs += IF(AND(g: _*) BOOL_== FALSE).THEN(Return(FALSE)).ENDIF
```

Comment:
there are some extra foldable boolean expressions ending up in bytecode compared to the old version:

```
    ICONST_1
    IFEQ L6
```

```
    ICONST_0
    IF_ICMPNE L42
    ICONST_0
    IRETURN
```

or in cfr syntax

```
        if (!true) return false;
```
```
        boolean bl2 = true;
        if (!bl2) {
            return false;
        }
```